Xamarin.Forms.Platform.UAP/VisualElementPackager.cs

```diff
@@ -119,8 +119,7 @@ void OnChildAdded(object sender, ElementEventArgs e)
 
 			_panel.Children.Add(childRenderer.ContainerElement);
 
-			if (ElementController.LogicalChildren[ElementController.LogicalChildren.Count - 1] != view)
-				EnsureZIndex();
+			EnsureZIndex();
```

Comment:
We don't want to just update the ZIndex's without cause because that will unwind this PR
https://github.com/xamarin/Xamarin.Forms/pull/1196

Which introduced the initial issue that I was fixing. The ideal scenario is that Z-Index is just not set at all unless the user inserts a view somewhere else in the stack or things get moved around (post renderer)

I'm still looking into this one as well because ideally EnzureZIndex isn't called unless it has to be. If that makes sense

Reply:
Calling EnsureZIndex() will not break the earlier fix because the child elements are already in the correct order at the end of the ChildAdded() event handler.

I'd like to stress that this regression bug breaks **every single UWP app** that uses Xamarin Forms. Apps assume that the last element is always rendered on top of everything else and currently it's the easiest (only?) way to implement popup-like views.